tree/ntuple/v7/src/RFieldBase.cxx

```diff
@@ -560,13 +599,28 @@ ROOT::Experimental::RFieldBase::Create(const std::string &fieldName, const std::
                   result = std::make_unique<RClassField>(fieldName, canonicalType);
                }
             }
+         } else if (options.fEmulateUnknownTypes) {
+            assert(desc);
```

Comment:
This option can be set through the public API and then we don't have a descriptor, no?

Reply:
With [e2c3074](https://github.com/root-project/root/pull/17288/commits/e2c3074bf4a2a3af219060f45f24ceb2da346dfa) the user cannot directly call `RFieldBase::Create()` with `fEmulateUnknownTypes == true`, so this code path is only taken when called by `RFieldDescriptor::CreateField` (which does pass down the descriptor)